frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -214,6 +218,42 @@ export default class FirebaseInteractor {
     };
 
     intervention.update(object);
+    this.updateDaysActive();
+  }
+
+  async updateDaysActive() {
+    let userId = this.auth.currentUser?.uid;
+
+    let today = new Date(
+      new Date().getFullYear(),
+      new Date().getMonth(),
+      new Date().getDate()
+    );
+    let todayDayOfWeek = today.getDay();
+    let user = await this.getUser(undefined);
+    let daysActive = user.daysActive;
+    let clearPreviousWeek = false;
+    daysActive.forEach((dayString) => {
+      let day = new Date(dayString);
+      let daysApart = (today.getTime() - day.getTime()) / (1000 * 60 * 60 * 24);
+      let dayOfWeek = day.getDay();
+      if (today !== day && (dayOfWeek > todayDayOfWeek || daysApart >= 7)) {
+        clearPreviousWeek = true;
+      }
+    });
+
+    if (clearPreviousWeek) {
+      await this.db.collection("users").doc(userId).update({
+        daysActive: [],
+      });
+    }
```

Comment:
Do we inherently need to clear the previous week? I think it would be better to keep the values so that we can use the previous week/month's usage for stats/insights about app usage. If we did not clear it, would that break the current implementation?

Reply:
We were thinking that storing all the dates for every user would be a lot of data to store. Would you suggest clearing after a month? Right now if we did not clear it it would break the current implementation, but it's something we could do.

Reply:
I honestly don't think we need to clear it at all. It probably just going to be a small string, so probably nothing to worry about